mne/realtime/client.py

```diff
@@ -364,7 +363,8 @@ def raw_buffers(self, nchan):
         raw_buffer : generator
             Generator for iteration over raw buffers.
         """
-        while True:
+        # stops the loop if stop_receive_thread has been called
+        while self._recv_thread == threading.current_thread():
```

Comment:
This should catch the (admittedly unlikely) case that ``stop_receive_thread`` and ``start_receive_thread`` are called while ``read_raw_buffer`` blocks the old thread, which would cause two receive threads to be running concurrently... If you prefer, I can change it to ``None``